lib/ld-eventsource/impl/buffered_line_reader.rb

```diff
@@ -0,0 +1,73 @@
+
+module SSE
+  module Impl
+    class BufferedLineReader
```

Comment:
For this particular case and at this point in time this likely won't be an issue, but singleton-classes can be a bit harder to work with and end up with run-on logic as things get added to them.

Reply:
There's a specific reason that this code was moved to its own class: to allow it to be unit-tested outside of the context of SSE::Client. I'd rather not get into hypothetical concerns of every possible way someone might overcomplicate this code in the future.